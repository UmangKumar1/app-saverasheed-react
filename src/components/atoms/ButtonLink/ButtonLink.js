import React from 'react'
import PropTypes from 'prop-types'

import Button from '../Button'
import Link from '../Link'

const { any, func, string } = PropTypes

const ButtonLink = ({
  to,
  children,
  btnClassName,
  btnWrapperClassName,
  ...props
}) =>
  to ? (
    <Link className={btnWrapperClassName} to={to} {...props}>
      <Button className={btnClassName}>{children}</Button>
    </Link>
  ) : (
    <span className={btnWrapperClassName} {...props}>
      <Button className={btnClassName}>{children}</Button>
    </span>
  )

ButtonLink.propTypes = {
  children: any.isRequired,
  onClick: func,
  to: string,
}

export default ButtonLink
