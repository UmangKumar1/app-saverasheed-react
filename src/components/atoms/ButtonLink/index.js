export { default, default as ButtonLink } from './ButtonLink'
