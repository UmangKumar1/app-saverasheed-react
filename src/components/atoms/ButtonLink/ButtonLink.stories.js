import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import ButtonLink from '.'

storiesOf('ButtonLink', module).add('default', () => (
  <ButtonLink onClick={action('click!')}>Click me!</ButtonLink>
))
