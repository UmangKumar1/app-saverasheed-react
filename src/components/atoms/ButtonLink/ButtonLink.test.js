import React from 'react'
import { shallow } from 'enzyme'
import ButtonLink from '.'

const wrap = (props = {}) => shallow(<ButtonLink {...props} />)

describe('ButtonLink', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ children: 'ButtonLink', onClick: jest.fn() })
    expect(wrapper).toMatchSnapshot()
  })
})
