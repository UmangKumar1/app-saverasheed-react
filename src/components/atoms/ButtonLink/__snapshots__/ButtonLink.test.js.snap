// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonLink
    onClick={[MockFunction]}
  >
    ButtonLink
  </ButtonLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Styled(button)>
        ButtonLink
      </Styled(button)>,
      "className": undefined,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "ButtonLink",
        "className": undefined,
      },
      "ref": null,
      "rendered": "ButtonLink",
      "type": [Function],
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Styled(button)>
          ButtonLink
        </Styled(button)>,
        "className": undefined,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "ButtonLink",
          "className": undefined,
        },
        "ref": null,
        "rendered": "ButtonLink",
        "type": [Function],
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
